app/assets/javascripts/countdown.js

```diff
@@ -0,0 +1,38 @@
+function Countdown(trackingElements) {
+  this.trackingElements = trackingElements;
+  this.auction_end_time = trackingElements.auction_end_time.text();
+
+  this.ongoing = function(durationCalculator) {
+    this.trackingElements.days.html(durationCalculator.days);
+    this.trackingElements.hours.html(durationCalculator.hours);
+    this.trackingElements.minutes.html(durationCalculator.minutes);
+    this.trackingElements.seconds.html(durationCalculator.seconds);
+  };
+
+  this.onEnd = function(durationCalculator) {
+    this.trackingElements.auction_end_time.html("Auction for this item is closed");
+    this.trackingElements.buy_now_button.hide();
+    this.trackingElements.bid_now_form.hide();
```

Comment:
The tracking elements parameter shouldn't know about bidding buttons and forms. This should be handled in the callback passed to onEnd wherever you use the timer.